### PR TITLE
feat: replace select picker v2

### PR DESCRIPTION
**Pull request by shivani170**
# Description

- [x] 1. ciConfig > CISelfDockerBuildOption.tsx
- [x] 2. ciPipeline > Webhook/WebhookDetailsModal.tsx
- [x] 3. ciConfig > WebhookSelectorCondition
- [x] ciConfig > ConditionContainer.tsx
     - [x] 4. Condition On variable
     - [x] 5. Assigned Operators
- [x] 6. CD config > Custom Image Tag
- [x] 7. SourceMaterial
- [x]  ciPipeline> CustomInputOutputVariable.tsx
     - [x] 8. Input Variable format
     - [x] 9. Output Variable format
- [x]  Docker.tsx > Registry Provider
- [x]  13.  CD Sidebar.tsx > Config Map & secret
- [x]  14.  -PluginDetailsHeader~
- [x]  15. Cluster > kubeconfig >**UseNameDropdown**
- [x]  16. NodeListSearchFilter > k8s version filter
- [x]  17. ClusterNodes/NodeActions/EditTaintsModal.tsx

Impacted areas:
app/Overview/Overview.tsx => Provider Icon replaced

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/CIPipelineN/ConditionContainer.tsx [hunk 1]
```diff
@@ -157,7 +165,8 @@ export const ConditionContainer = ({ type }: { type: ConditionContainerType }) =
         setFormData(_formData)
     }
 
-    const handleConditionOnVariableChange = (selectedValue: { label: string; value: number }, index: number): void => {
+    const handleConditionOnVariableChange = (selectedValue: OptionType, index: number): void => {
+        console.log(selectedValue)
```
Comment (RohitRaj011) [inline src/components/CIPipelineN/ConditionContainer.tsx:169]: remove this

File: src/components/CIPipelineN/ConditionContainer.tsx [hunk 2]
```diff
@@ -337,9 +348,8 @@ export const ConditionContainer = ({ type }: { type: ConditionContainerType }) =
                                                       }
                                                     : null
                                             }
-                                            tabIndex={1}
                                             placeholder="Select variable"
-                                            onChange={(selectedValue) => {
+                                            onChange={(selectedValue: OptionType) => {
```
Comment (RohitRaj011) [inline src/components/CIPipelineN/ConditionContainer.tsx:352]: extract this function out

File: src/components/CIPipelineN/ConditionContainer.tsx [hunk 3]
```diff
@@ -352,39 +362,29 @@ export const ConditionContainer = ({ type }: { type: ConditionContainerType }) =
...
                                                 conditionDetail.conditionOperator
                                                     ? {
                                                           label: conditionDetail.conditionOperator,
                                                           value: conditionDetail.conditionOperator,
                                                       }
                                                     : selectedOperator
                                             }
-                                            tabIndex={1}
-                                            onChange={(selectedValue) => {
+                                            onChange={(selectedValue: OptionType) =>
```
Comment (RohitRaj011) [inline src/components/CIPipelineN/ConditionContainer.tsx:380]: same

File: src/components/CIPipelineN/Sidebar.tsx [hunk 4]
```diff
@@ -193,8 +175,7 @@ export const Sidebar = ({
             addConfigSecrets(updatedList)
         }
 
-        const searchGroupHeading = (props) => <GroupHeading {...props} hideClusterName />
-
+        console.log('configMapAndSecrets', configMapAndSecrets)
```
Comment (RohitRaj011) [inline src/components/CIPipelineN/Sidebar.tsx:178]: remove

File: src/components/ciConfig/CIBuildpackBuildOptions.tsx [hunk 5]
```diff
@@ -71,6 +72,53 @@ export const renderOptionIcon = (option: string) => {
...
+        return {
+            label: selectedMaterial.name,
+            value: selectedMaterial.checkoutPath,
+            startIcon: getGitProviderIcon(selectedMaterial.url),
+        }
+    }
+    return null
+}
+
+// TODO: Remove console after testing,
```
Comment (RohitRaj011) [inline src/components/ciConfig/CIBuildpackBuildOptions.tsx:96]: remove if not required

**Review by RohitRaj011 — CHANGES_REQUESTED** [review]
please do self-review once 

multiple console.logs and comments remaining

File: src/components/CIPipelineN/ConditionContainer.tsx (hunk 2 above)
Comment (shivani170) [reply src/components/CIPipelineN/ConditionContainer.tsx:352]: It is extracted out already, To pass index have to do this way.

File: src/components/ciConfig/CIBuildpackBuildOptions.tsx (hunk 5 above)
Comment (shivani170) [reply src/components/ciConfig/CIBuildpackBuildOptions.tsx:96]: moved it to another pr

File: src/components/ciPipeline/types.ts [hunk 6]
```diff
@@ -412,3 +413,13 @@ export enum CIPipelineBuildType {
     LINKED_CD = 'LINKED_CD',
     NORMAL_JOB = 'NORMAL_JOB',
 }
+
+export interface WebhookConditionType {
+    conditionIndex: number
+    masterSelectorList: OptionType[]
+    selectorCondition: any
```
Comment (RohitRaj011) [inline src/components/ciPipeline/types.ts:420]: suggestion: add proper type

File: src/css/base.scss [hunk 7]
```diff
@@ -2598,6 +2598,10 @@ button.anchor {
     border-top-right-radius: 3px;
 }
 
+.dc__top-right-radius-4 {
```
Comment (RohitRaj011) [inline src/css/base.scss:2601]: class already exists

`.dc__top-right-radius`

**Review by RohitRaj011 — CHANGES_REQUESTED** [review]

**Review by RohitRaj011 — DISMISSED** [review]

**Review by RohitRaj011 — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**